src/codegen/mod.rs

```diff
@@ -360,6 +364,12 @@ impl CodeGenerator for Module {
                 }
             }
 
+            if item.id() == ctx.sentinel_module() {
```

Comment:
The whole `sentinel_module` thing really feels like a hack. Is it just for this? If so, if you always want this at the top of the file, let's just move its handling above.